### PR TITLE
Provide a different message if you received credits for negative maintenance costs

### PR DESCRIPTION
**Pull request by Amazinite**
Ties up a loose end with #6423 that I realized the moment I merged it.

### CODE REVIEW

**Review by Zitchas — APPROVED** [review]
To my very limited ability to read C++, this looks good.

File: source/Account.cpp [hunk 1]
```diff
@@ -224,6 +224,19 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
...
+	auto creditString = [](int64_t payment) -> string
+	{
+		return payment == 1 ? "1 credit" : Format::Credits(payment) + " credits";
+	};
+	
+	if(maintenancePaid < 0)
+	{
+		if(missedPayment)
+			out << " ";
+		out << "You received " << creditString(-maintenancePaid) << " in maintenance.";
```
Comment (petervdmeer) [inline source/Account.cpp:236]: Maintenance only makes money when you are the person doing the maintenance, which is not the case here.

Maybe reword so something like:
```suggestion
		out << "Your ships and outfits netted you " << creditString(-maintenancePaid) << ".";
```

File: source/Account.cpp [hunk 2]
```diff
@@ -224,6 +224,19 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
...
+		maintenancePaid = 0;
+	}
+	
 	// If you didn't make any payments, no need to continue further.
 	if(!(salariesPaid + maintenancePaid + mortgagesPaid + finesPaid))
```
Comment (petervdmeer) [inline source/Account.cpp:241]: MaintenancePaid could now cancel out the others and doesn't need to be handled when negative;

```suggestion
	if(!salariesPaid && (maintenancePaid <= 0) && !mortgagesPaid &&!finesPaid)
```

File: source/Account.cpp [hunk 3]
```diff
@@ -232,11 +245,6 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
...
 	out << "You paid ";
 	
-	auto creditString = [](int64_t payment) -> string
-	{
-		return payment == 1 ? "1 credit" : Format::Credits(payment) + " credits";
-	};
-	
 	map<string, int64_t> typesPaid;
 	if(salariesPaid)
 		typesPaid["crew salaries"] = salariesPaid;
```
Comment (petervdmeer) [inline source/Account.cpp:250]: The line below should be `if(maintenancePaid > 0)`, but GitHub doesn't allow me to write a comment on that line.

File: source/Account.cpp [hunk 4]
```diff
@@ -265,7 +273,7 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
 			out << creditString(salariesPaid) << " in crew salaries"
 				<< ((mortgagesPaid || finesPaid || maintenancePaid) ? " and " : ".");
 		if(maintenancePaid)
```
Comment (petervdmeer) [inline source/Account.cpp:275]:
```suggestion
		if(maintenancePaid > 0)
```

File: source/Account.cpp [hunk 5]
```diff
@@ -265,7 +273,7 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
 			out << creditString(salariesPaid) << " in crew salaries"
 				<< ((mortgagesPaid || finesPaid || maintenancePaid) ? " and " : ".");
```
Comment (petervdmeer) [inline source/Account.cpp:274]:
```suggestion
				<< ((mortgagesPaid || finesPaid || (maintenancePaid > 0)) ? " and " : ".");
```

**Review by petervdmeer — COMMENTED** [review]
Sorry for creating such an incomplete PR. I have some more comments to make the negative costs feature complete.

File: source/Account.cpp (hunk 1 above)
Comment (petervdmeer) [reply source/Account.cpp:236]: > Maintenance only makes money when you are the person doing the maintenance, which is not the case here.

Or reword to: "You received (credits) based on your ships and outfits."

**Comment by tehhowch** [comment]
I would much prefer we revert #6423 and use a different attribute entirely, rather than always need to handle a positive and negative value for a "cost."

Additionally, a separate attribute allows the player to see the incoming and outgoing separately, rather than just the net change.

Going even further, if we were to ever support income / cost scaling (e.g. in a difficulty mode), we would have to check and scale each contribution rather than the final sum.

Example attributes would use something like "stipend," "allowance," "affordance," "per diem," "allotment," etc.

**Comment by petervdmeer** [comment]
> I would much prefer we revert #6423 and use a different attribute entirely, rather than always need to handle a positive and negative value for a "cost."

Entirely different attributes would indeed be cleaner. I would not disagree to reverting to allow the feature to be re-implemented using separate keywords.

> Additionally, a separate attribute allows the player to see the incoming and outgoing separately, rather than just the net change.

We could also achieve this with the "overloaded" attributes, but separate attributes is indeed more clean and more in line with existing code.

> Going even further, if we were to ever support income / cost scaling (e.g. in a difficulty mode), we would have to check and scale each contribution rather than the final sum.
> Example attributes would use something like "stipend," "allowance," "affordance," "per diem," "allotment," etc.

Let's discuss the proper keywords for the feature in the original ticket #6406 .

File: source/Account.cpp (hunk 2 above)
Comment (Amazinite) [reply source/Account.cpp:241]: That's why I set maintenancePaid to 0 after saying how much you received so that we don't need to handle the negative case for payments.
